### PR TITLE
Issue 65 - data validation on contact form, company and facility detail forms

### PR DESCRIPTION
**Pull request by RLiNREL**
Implemented data validation
#114 added validation to company name and zip code
#115 added validation to name (first and last name), phone number, and email
#117 same with 114

### CODE REVIEW

**Comment by jordanperr** [comment]
I tested the latest version v0.0.1-alpha-03c6da6.
- Zip code validation does not seem to be implemented.
- "New Company" should be removed as a default value from company name field. Just leave it empty.
- Badges in the sidebar have not yet been merged into this PR.

Perhaps out of scope for this pull request, I noticed the other fields in the General Information tab are not yet validated.

File: src/app/setup-wizard/pre-visit/company-contacts-setup/company-contacts-form/company-contacts-form.service.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    let lastname = contactForm.controls['lastname'].value;
+    contact.name = '';
+    if (firstname || lastname) {
+      contact.name = firstname + ', ' + lastname;
+    }
+    //TODO: add all the properties that will get updated by the form
+    let phone = contactForm.controls['phone'].value;
+    let ext = contactForm.controls['ext'].value;
+    const phoneWithExt = phone + (ext? ' ext. ' + ext : '');
+    // console.log('phone:', phone, 'ext:', ext, 'phext:', phoneWithExt);
```
Comment (rmroot) [inline src/app/setup-wizard/pre-visit/company-contacts-setup/company-contacts-form/company-contacts-form.service.ts]: Please remove commented out debugging logs

**Review by rmroot — CHANGES_REQUESTED** [review]
Overall looks good. Please remove commented out debugging code

File: src/app/setup-wizard/pre-visit/company-contacts-setup/company-contacts-form/company-contacts-form.service.ts (hunk 1 above)
Comment (RLiNREL) [reply src/app/setup-wizard/pre-visit/company-contacts-setup/company-contacts-form/company-contacts-form.service.ts]: Thanks! They have been removed.

**Review by rmroot — APPROVED** [review]

**Pull request merged**